docker/templates/devel.template

```diff
@@ -0,0 +1,42 @@
+RUN apt-get update \
+ && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    libgflags-dev \
+    libgoogle-glog-dev \
+    libopencv-dev \
+    libleveldb-dev \
+    libsnappy-dev \
+    liblmdb-dev \
+    libhdf5-serial-dev \
+    libprotobuf-dev \
+    protobuf-compiler \
+    libatlas-base-dev \
+    python-dev \
+    python-pip \
+    python-numpy \
+    python-scipy \
+    gfortran
+
+# Use --no-install-recommends for the boost libs.
+RUN apt-get install -y --no-install-recommends \
+    libboost-all-dev
+
+# Install utilities used for the build.
+RUN apt-get install -y \
+    build-essential \
+    cmake \
+    cmake-curses-gui \
+    git \
+    ca-certificates \
+    bc \
+    wget
+
+# Install the python requirements.
+RUN wget https://raw.githubusercontent.com/BVLC/caffe/master/python/requirements.txt -O /tmp/requirements.txt && \
```

Comment:
As I mentioned in response to one of @fix42's comments, the development
image is intended to be used to build caffe from source locally (with
folders mounted as volumes. See `start_caffe_docker.sh`  for example). This
means that the caffe source (and thus requirements.txt) is not available to
the image at build time. You will note that the runtime Dockerfiles (and
their corresponding templates) do use the locally available requirements
files.

Of course, if someone has a suggestion as to how this could be improved,
I'm all ears.
-- 
Evan Lezar
www.evanlezar.com